src/ressources/customers/AvatarField.tsx

```diff
@@ -0,0 +1,24 @@
+import { Avatar } from "@mui/material";
+import { RaRecord, UseRecordContextParams } from "react-admin";
+
+export const AvatarField = ({
+  record,
+}: UseRecordContextParams): JSX.Element | null => {
+  if (!record || (!record.avatar && !record.last_name && !record.first_name))
+    return null;
+
+  if (record.avatar) return <Avatar src={record.avatar} />;
+
+  return <Avatar {...stringAvatar(record)} />;
+};
+
+function stringAvatar(record: RaRecord) {
```

Comment:
very weird. Why is it in another function?

Reply:
Can you explain why? Is it because I return null with the case without first name and last name?

I'm using this from mui doc, but maybe I'm dealing wrong with the case without first name and last name...
![image](https://user-images.githubusercontent.com/100569076/162393140-070dac96-4003-41d3-81b3-cb4bb56012d4.png)

https://mui.com/components/avatars/#letter-avatars

Reply:
Indeed it took me some time to figure out this code actually handles the avatar for customer who do not have an avatar picture. Maybe it would be more obvious if you renamed the `stringAvatar()` function to `letterAvatar()`?

But anyway, your code is fine, let's keep it!